### PR TITLE
[WFLY-9106]: Hard-code default values in messaging-activemq subsystem.

### PR DESCRIPTION
**Pull request by ehsavoie**
Replacing ActiveMQClient., ActiveMQDefaultConfiguration default values usage by hardcoded values in the messaging subsystem.

JIRA: https://issues.jboss.org/browse/WFLY-9106

### CODE REVIEW

**Review by kabir — DISMISSED** [review]
@ehsavoie There is a checkstyle problem

**Comment by kabir** [comment]
[amq-master.txt](https://github.com/wildfly/wildfly/files/2396553/amq-master.txt)
[amq-pull.txt](https://github.com/wildfly/wildfly/files/2396555/amq-pull.txt)
@ehsavoie amq-master.txt contains a r-r-d for the subsystem built from master. amq-pull.txt contains r-r-d built from this PR. There are some differences.

{code}
$diff -u amq-master.txt amq-pull.txt 
--- amq-master.txt	2018-09-19 11:06:06.000000000 +0100
+++ amq-pull.txt	2018-09-19 11:04:39.000000000 +0100
@@ -1120,7 +1120,7 @@
                             "expressions-allowed" => true,
                             "required" => false,
                             "nillable" => true,
-                            "default" => 10485760,
+                            "default" => 10485760L,
                             "unit" => "BYTES",
                             "min" => 1024L,
                             "max" => 9223372036854775807L,
@@ -1273,7 +1273,7 @@
                             "expressions-allowed" => true,
                             "required" => false,
                             "nillable" => true,
-                            "default" => -1,
+                            "default" => 10,
                             "access-type" => "read-write",
                             "storage" => "configuration",
                             "restart-required" => "all-services"
@@ -2241,7 +2241,7 @@
                                         "expressions-allowed" => true,
                                         "required" => false,
                                         "nillable" => true,
-                                        "default" => 1.0,
+                                        "default" => big decimal 1,
                                         "access-type" => "read-write",
                                         "storage" => "configuration",
                                         "restart-required" => "all-services"
@@ -2600,7 +2600,7 @@
                                             "expressions-allowed" => true,
                                             "required" => false,
                                             "nillable" => true,
-                                            "default" => 5000L,
+                                            "default" => 5000,
                                             "unit" => "MILLISECONDS",
                                             "access-type" => "read-write",
                                             "storage" => "configuration",
@@ -2691,7 +2691,7 @@
                                                         "expressions-allowed" => true,
                                                         "required" => false,
                                                         "nillable" => true,
-                                                        "default" => 30000L,
+                                                        "default" => 30000,
                                                         "unit" => "MILLISECONDS",
                                                         "access-type" => "read-write",
                                                         "storage" => "configuration",
@@ -2834,7 +2834,7 @@
                                                         "expressions-allowed" => true,
                                                         "required" => false,
                                                         "nillable" => true,
-                                                        "default" => 30000L,
+                                                        "default" => 30000,
                                                         "unit" => "MILLISECONDS",
                                                         "access-type" => "read-write",
                                                         "storage" => "configuration",
@@ -2892,7 +2892,7 @@
                                             "expressions-allowed" => true,
                                             "required" => false,
                                             "nillable" => true,
-                                            "default" => 30000L,
+                                            "default" => 30000,
                                             "unit" => "MILLISECONDS",
                                             "access-type" => "read-write",
                                             "storage" => "configuration",
@@ -3145,7 +3145,7 @@
                                             "expressions-allowed" => true,
                                             "required" => false,
                                             "nillable" => true,
-                                            "default" => 30000L,
+                                            "default" => 30000,
                                             "unit" => "MILLISECONDS",
                                             "access-type" => "read-write",
                                             "storage" => "configuration",
@@ -3275,7 +3275,7 @@
                                             "expressions-allowed" => true,
                                             "required" => false,
                                             "nillable" => true,
-                                            "default" => 5000L,
+                                            "default" => 5000,
                                             "unit" => "MILLISECONDS",
                                             "access-type" => "read-write",
                                             "storage" => "configuration",
@@ -4089,7 +4089,7 @@
                                         "expressions-allowed" => true,
                                         "required" => false,
                                         "nillable" => true,
-                                        "default" => 1.0,
+                                        "default" => big decimal 1,
                                         "access-type" => "read-write",
                                         "storage" => "configuration",
                                         "restart-required" => "all-services"
@@ -5441,7 +5441,7 @@
                                         "expressions-allowed" => true,
                                         "required" => false,
                                         "nillable" => true,
-                                        "default" => 1.0,
+                                        "default" => big decimal 1,
                                         "access-type" => "read-write",
                                         "storage" => "configuration",
                                         "restart-required" => "all-services"
@@ -5916,7 +5916,7 @@
                                         "expressions-allowed" => true,
                                         "required" => false,
                                         "nillable" => true,
-                                        "default" => 1.0,
+                                        "default" => big decimal 1,
                                         "access-type" => "read-write",
                                         "storage" => "configuration",
                                         "restart-required" => "all-services"
@@ -5991,7 +5991,7 @@
                                         "expressions-allowed" => true,
                                         "required" => false,
                                         "nillable" => true,
-                                        "default" => 2000L,
+                                        "default" => 2000,
                                         "unit" => "MILLISECONDS",
                                         "access-type" => "read-write",
                                         "storage" => "configuration",
@@ -7534,7 +7534,7 @@
                             "expressions-allowed" => true,
                             "required" => false,
                             "nillable" => true,
-                            "default" => 1.0,
+                            "default" => big decimal 1,
                             "access-type" => "read-write",
                             "storage" => "configuration",
                             "restart-required" => "all-services"
{code}

There seem to be a few sets of discrepancies:
1) Values becoming longs (e.g. 10485760L rather than 10485760) is probably ok
2) Values becoming ints (e.g. 5000L rather than 5000) is probably ok, but doesn't feel consistent.
3) Things which were -1 becoming 10 are probably not ok? 
4) 'big decimal 1' just looks weird :)

**Comment by ehsavoie** [comment]
1. Please note that the type of the attribute is LONG 
2. is wrong
3. it is correct, please note that we set the value to 10 in our configurations
4. 'big decimal 1' just looks weird :) can't agree more but it is the type defined in the subsystem BIG_DECIMAL. I could use a simple double value for it.

**Comment by jmesnil** [comment]
failures are related to the PR

**Review by jmesnil — CHANGES_REQUESTED** [review]
I don't understand why the model version and XSD had to change to replace Artemis constants with hard-code values?

File: galleon-pack/src/main/resources/feature_groups/messaging-activemq-ha.xml [hunk 1]
```diff
@@ -5,7 +5,6 @@
         <feature spec="subsystem.messaging-activemq.server">
             <param name="server" value="default"/>
             <param name="cluster-password" value="${jboss.messaging.cluster.password:CHANGE ME!!}"/>
-            <param name="journal-pool-files" value="10"/>
```
[the remark this answers is not in the document]
Comment (ehsavoie) [reply galleon-pack/src/main/resources/feature_groups/messaging-activemq-ha.xml]: this value is the default one, so removing it makes sense

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingTransformerRegistration.java [hunk 2]
```diff
@@ -68,11 +72,239 @@ public String getSubsystemName() {
...
         registerTransformers_EAP_7_2_0(builder.createBuilder(MessagingExtension.VERSION_4_0_0, MessagingExtension.VERSION_3_0_0));
         registerTransformers_EAP_7_1_0(builder.createBuilder(MessagingExtension.VERSION_3_0_0, MessagingExtension.VERSION_2_0_0));
         registerTransformers_EAP_7_0_0(builder.createBuilder(MessagingExtension.VERSION_2_0_0, MessagingExtension.VERSION_1_0_0));
 
-        builder.buildAndRegister(registration, new ModelVersion[] { MessagingExtension.VERSION_1_0_0, MessagingExtension.VERSION_2_0_0, MessagingExtension.VERSION_3_0_0 });
+        builder.buildAndRegister(registration, new ModelVersion[] { MessagingExtension.VERSION_1_0_0, MessagingExtension.VERSION_2_0_0, MessagingExtension.VERSION_3_0_0 , MessagingExtension.VERSION_4_0_0});
+    }
+
+    private static void registerTransformers_WF_15(ResourceTransformationDescriptionBuilder subsystem) {
+        ResourceTransformationDescriptionBuilder server = subsystem.addChildResource(MessagingExtension.SERVER_PATH);
```
[the remark this answers is not in the document]
Comment (ehsavoie) [reply messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingTransformerRegistration.java]: ok

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ServerDefinition.java [hunk 3]
```diff
@@ -542,7 +542,7 @@
     public static final SimpleAttributeDefinition MANAGEMENT_ADDRESS = create("management-address", ModelType.STRING)
             .setAttributeGroup("management")
             .setXmlName("address")
-            .setDefaultValue(new ModelNode(ActiveMQDefaultConfiguration.getDefaultManagementAddress().toString()))
+            .setDefaultValue(new ModelNode(new SimpleString("activemq.management").toString()))
```
[the remark this answers is not in the document]
Comment (ehsavoie) [reply messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/ServerDefinition.java]: ok

File: messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingExtension.java [hunk 4]
```diff
@@ -164,13 +171,14 @@
 
     static final String RESOURCE_NAME = MessagingExtension.class.getPackage().getName() + ".LocalDescriptions";
 
+    protected static final ModelVersion VERSION_5_0_0 = ModelVersion.create(5, 0, 0);
     protected static final ModelVersion VERSION_4_0_0 = ModelVersion.create(4, 0, 0);
     protected static final ModelVersion VERSION_3_0_0 = ModelVersion.create(3, 0, 0);
     protected static final ModelVersion VERSION_2_0_0 = ModelVersion.create(2, 0, 0);
     protected static final ModelVersion VERSION_1_0_0 = ModelVersion.create(1, 0, 0);
-    private static final ModelVersion CURRENT_MODEL_VERSION = VERSION_4_0_0;
```
[the remark this answers is not in the document]
Comment (ehsavoie) [reply messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingExtension.java]: Ok, will do the version bump in a separate PR

[the remark this answers is not in the document]
Comment (ehsavoie) [reply messaging-activemq/src/main/java/org/wildfly/extension/messaging/activemq/MessagingExtension.java]: Ok, will do the version bump in a separate PR https://github.com/wildfly/wildfly/pull/11678

**Comment by ehsavoie** [comment]
Everything should be fixed

**Review by jmesnil — APPROVED** [review]

**Comment by kabir** [comment]
@ehsavoie needs a rebase

**Pull request merged**